### PR TITLE
[WIP] Fix for OpenMP library clashing on MacOS

### PR DESCRIPTION
**Pull request by atilag**
### Summary
Fixes Issue #1
 

### Details and comments
On Mac, OpenMP implementation is integrated through an external library: `libomp.dylib` (and some others similar like libiomp5 from Intel). This library has a constraint: It cannot be loaded twice in memory, basically because it will imply a double initialization of it's runtime, and this is potentially dangerous. We do need this library to operate, but other depedencies we have, require them as well, which is a problem because of the way Python loads extensions (and we, are an extension too).
Basically, with this fix, we have provided Aer the ability to dynamcally load libomp in every possible scenario.

### CODE REVIEW

**Comment by atilag** [comment]
I'm closing this because I've messed things up while squashing/rebasing some commits and the history is dirty
